iceoryx_utils/cmake/IceoryxPlatformDetection.cmake

```diff
@@ -22,11 +22,11 @@ if(UNIX AND NOT APPLE)
 endif(UNIX AND NOT APPLE)
 
 if(LINUX)
-    set(ICEORYX_CXX_STANDARD 11)
+    set(ICEORYX_CXX_STANDARD 14)
```

Comment:
I just ran grep and found 2 remaining `std=c++11` settings in:
- iceoryx_dds/CMakeLists.txt
- iceoryx_utils/testutils/compile_test.hpp:

Reply:
there where also a few other occurrences were `set_target_properties(... CXX_STANDARD 11)` was used